src/Http/Controllers/U2fController.php

```diff
@@ -35,90 +29,83 @@ public function __construct(LaravelU2f $u2f, Config $config)
-            Event::fire('u2f.register', ['u2fKey' => $key, 'user' => Auth::user()]);
-            Session::forget('u2f.registerData');
-
+            $key = $this->u2f->doRegister($user, session()->put('u2f.registerData'), json_decode($request->input('register')));
+            event('u2f.register', ['u2fKey' => $key, 'user' => $user]);
+            session()->forget('u2f.registerData');
             if ($this->config->get('u2f.register.postSuccessRedirectRoute')) {
-                return Redirect::route($this->config->get('u2f.register.postSuccessRedirectRoute'));
+                return redirect()->route($this->config->get('u2f.register.postSuccessRedirectRoute'));
             }
-
             return redirect('/');
         } catch (Exception $e) {
-            return Redirect::route('u2f.register.data');
+            return redirect()->route('u2f.register.data');
         }
     }
 
     /**
-     * @author LAHAXE Arnaud
-     *
-     * @return mixed
-     */
-    public function authData()
+    * @param  Request $request
+    * @return mixed
+    */
+    public function authData(Request $request)
     {
-        if ($this->u2f->check()) {
-            return $this->redirectAfterSuccessAuth();
-        }
-
-        $req = $this->u2f->getAuthenticateData(Auth::user());
-        Event::fire('u2f.authentication.data', ['user' => Auth::user()]);
+        $user = $request->user();
+        if ($this->u2f->check()) return $this->redirectAfterSuccessAuth();
 
-        Session::set('u2f.authenticationData', $req);
+        $req = $this->u2f->getAuthenticateData($user);
+        event('u2f.authentication.data', ['user' => $user]);
+        session()->put('u2f.authenticationData', $req);
 
         return view($this->config->get('u2f.authenticate.view'))
             ->with('authenticationData', $req);
     }
 
     /**
-     * @author LAHAXE Arnaud
-     *
-     * @return mixed
-     */
-    public function auth()
+    * @param  Request $request
+    * @return mixed
+    */
+    public function auth(Request $request)
     {
+        $user = $request->user();
         try {
-            $key = $this->u2f->doAuthenticate(Auth::user(), Session::get('u2f.authenticationData'), json_decode(Input::get('authentication')));
-            Event::fire('u2f.authentication', ['u2fKey' => $key, 'user' => Auth::user()]);
-            Session::forget('u2f.authenticationData');
-
+            $key = $this->u2f->doAuthenticate($user, session()->put('u2f.authenticationData'), json_decode($request->input('authentication')));
+            event('u2f.authentication', ['u2fKey' => $key, 'user' => $user]);
+            session()->forget('u2f.authenticationData');
             return $this->redirectAfterSuccessAuth();
         } catch (Exception $e) {
-            Session::flash('error', $e->getMessage());
-
-            return Redirect::route('u2f.auth.data');
+            //Session::flash('error', $e->getMessage());
```

Comment:
Could you uncomment this and replace it with the session method on the request object?

```php
$request->session()->flash('error', $e->getMessage());
```